src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -372,14 +378,37 @@ public async Task<int> GenerateAIReview(ClaimsPrincipal user, string reviewId, s
                 payload.Add("base", String.Join("\\n", diffCodeLines.Select(item => item.lineText.Trim())));
             }
 
-            var result = new AIReviewModel();
+            var result = new AIReviewJobPolledResponseModel();
             try {
-                var response = await client.PostAsync(url, new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json"));
+                var response = await client.PostAsync(startUrl, new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json"));
                 response.EnsureSuccessStatusCode();
                 var responseString = await response.Content.ReadAsStringAsync();
-                result = JsonSerializer.Deserialize<AIReviewModel>(responseString);
+                var jobStartedResponse = JsonSerializer.Deserialize<AIReviewJobStartedResponseModel>(responseString);
+                activeApiRevision.CopilotReviewInProgress = true;
+                await _apiRevisionsManager.UpdateAPIRevisionAsync(activeApiRevision);
+
+                var poolUrl = $"{copilotEndpoint}/api-review/{jobStartedResponse.JobId}";
+                var poller = new Poller();
+                result = await poller.PollAsync<AIReviewJobPolledResponseModel>(
+                    operation: async () => {
+                        var response = await client.GetAsync(poolUrl);
+                        response.EnsureSuccessStatusCode();
+                        var pollResponseString = await response.Content.ReadAsStringAsync();
+                        var pollResponse = JsonSerializer.Deserialize<AIReviewJobPolledResponseModel>(pollResponseString);
+                        return pollResponse;
+                    },
+                    isComplete: response => (response.Status != "InProgress"),
+                    initialInterval : 2,
+                    maxInterval: 2
+                );
```

Comment:
Yeah, this seems right. Not sure what that other polling logic is that goes up to 8 minutes.

Reply:
Yes, this override the exponential Backof default on the Poller class.